### PR TITLE
test: Refactored tests that were still using the tap mocha shim + chai to now use tap.

### PR DESCRIPTION
**Pull request by bizob2828**
## Description

This PR refactored the remaining tests that still relied on mocha + chai.  I also consolidated some of the custom test assertions into `test/lib/custom-assertions.js`.  Lastly, I refactored the metrics helpers to not longer use assert and/or chai and use tap. With all of that I was able to remove chai from the devDependency list.  This will also help developers triaging failulres in tests as they are now all tap assertions.  When relying on mocha + chai it was very difficult at times, or even worse relying on the built in assert package.

### CODE REVIEW

File: test/lib/agent_helper.js [hunk 1]
```diff
@@ -71,639 +65,572 @@ function FakeSegment(transaction, duration, name = 'FakeSegment') {
   }
 }
 
-const helper = (module.exports = {
```
Comment (jsumners-nr) [inline test/lib/agent_helper.js:74]: I really like that this sort of declaration has been removed. It's quite difficult to read and keep track of. I think the refactored declaration is also going to help quite a lot with jsdoc and ide integration. The only thing missing from the refactor is the removal of the singleton `_agent`.

File: test/lib/agent_helper.js [hunk 2]
```diff
@@ -71,639 +65,572 @@ function FakeSegment(transaction, duration, name = 'FakeSegment') {
...
+helper.withSSL = () => {
+  return Promise.all([fs.readFile(KEYPATH), fs.readFile(CERTPATH), fs.readFile(CAPATH)])
+}
+
+helper.randomPort = (callback) => {
+  const net = require('net')
+  // Min port: 1024 (without root)
+  // Max port: 65535
+  // Our range: 1024-65024
+  const port = Math.ceil(Math.random() * 64000 + 1024)
```
Comment (jsumners-nr) [inline test/lib/agent_helper.js:366]: For what it's worth, this could be:

```suggestion
  const port = require('crypto').randomInt(1024, 65024)
```

**Review by jsumners-nr — APPROVED** [review]
Fantastic work. I didn't read everything, but the pattern looked clear. Great job sticking to just the minimum necessary.

File: test/lib/agent_helper.js (hunk 1 above)
Comment (bizob2828) [reply test/lib/agent_helper.js:74]: yea this was something I wasn't planning on doing but had to edit a few things here so broke this down.  The agent singleton is going to be a heavier lift. I don't know the effects of this.  I can log a story in our cleanup feature for this

File: test/lib/agent_helper.js (hunk 2 above)
Comment (bizob2828) [reply test/lib/agent_helper.js:366]: I'd like to stick with this kind of refactor in a diff PR, I agree though. I'll queue that one up later

**Pull request merged**